src/markdown_generators.js

```diff
@@ -4,35 +4,66 @@ const yaml = require("js-yaml")
 const markdown = require("markdown-builder")
 const titleCase = require("title-case")
 const TurndownService = require("turndown")
+const turndownPluginGfm = require("turndown-plugin-gfm")
+const helpers = require("./helpers")
+const gfm = turndownPluginGfm.gfm
+const tables = turndownPluginGfm.tables
 const turndownService = new TurndownService()
+turndownService.use(gfm)
+turndownService.use(tables)
+
+const REPLACETHISWITHAPIPE = "REPLACETHISWITHAPIPE"
+const REFSHORTCODESTART = "REFSHORTCODESTART"
+const REFSHORTCODEEND = "REFSHORTCODEEND"
+
+/**
+ * Sanitize markdown table content
+ **/
 turndownService.addRule("table", {
   filter:      ["table"],
-  replacement: content => {
-    return `{{< rawhtml >}}<table>${content}</table>{{< /rawhtml >}}`
-  }
-})
-turndownService.addRule("th", {
-  filter:      ["th"],
-  replacement: content => {
-    return `<th>${content}</th>`
-  }
-})
-turndownService.addRule("tr", {
-  filter:      ["tr"],
-  replacement: content => {
-    return `<tr>${content}</tr>`
-  }
-})
-turndownService.addRule("td", {
-  filter:      ["td"],
-  replacement: content => {
-    return `<td>${content}</td>`
+  replacement: (content, node, options) => {
+    /**
+     * Interate the HTML node and replace all pipes inside table
+     * cells with a marker we'll use later
+     */
+    for (let i = 0; i < node.rows.length; i++) {
+      const cells = node.rows[i].cells
+      for (let j = 0; j < cells.length; j++) {
+        cells[j].innerHTML = cells[j].innerHTML.replace(
+          /\|/g,
+          REPLACETHISWITHAPIPE
+        )
+      }
+    }
+    // Regenerate markdown for this table with cell edits
+    content = turndownService.turndown(node)
+    /**
+     * Get the bounds of the table, remove all line breaks, then
+     * from earlier with the HTML character entity for a pipe
+     * reintroduce them between rows, replacing our pipe marker
+     */
+    content = content
+      .substring(content.indexOf("|"), content.lastIndexOf("|"))
+      .replace(/\r?\n|\r/g, "{{< br >}}")
+      .replace(/\|{{< br >}}\|/g, "|\n|")
+      .replace(/REPLACETHISWITHAPIPE/g, "&#124;")
+    // Only do this if header lines are found
+    if (content.match(/(?<=\| \*\*)(.*?)(?=\*\* \|\r?\n|\r)/g)) {
```

Comment:
some of these longer / more complicated regexes it might be a good idea to pull out as a separate viable and just writing a little comment describing what they're supposed to match